### PR TITLE
added x64dbg module

### PR DESCRIPTION
**Pull request by mrexodia**
This is the snowman module used in [x64dbg](http://x64dbg.com), I think it would be nice to have this in the main repository (there is no intrusion with the current source code, only additions).

### CODE REVIEW

**Comment by yegord** [comment]
Let's discuss this. My position so far is that plugins to other tools should be in the source tree of the other tools. There are several reasons behind this.

The first reason is that, if I accept this patch, I will have to accept every other plugin people with create using the decompiler. And it is a frightening perspective.

The second reason is, I will have to do a code review of each change to every plugin, and it will take quite some time, both mine and the time of the change's author.

The third reason is, the author will have to obey the coding rules of the project, write the documentation (e.g., build instructions for his module), and I will have to check that all.

The fourth reason is, that I will have to support continuous integration for that plugin too, which will again take more of my time, without any profit.

That's why I would prefer people to use Snowman/libnc-* as a library, just copied into the source tree of the project using it, or being used as a git submodule. If Snowman's source tree must be modified to better support this scenario, I will be glad to accept a corresponding patch.

**Comment by mrexodia** [comment]
Okay, sounds fair to me :) I'll just leave the module in my repository and merge the master of this repository in it every time things change over here.

**Comment by yegord** [comment]
Yes, forking and rebasing/merging from time to time is another sensible option.
Thank you for understanding. :)

File: src/x64dbg-module/SnowmanView.cpp [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+    virtual nc::ByteSize readBytes(nc::ByteAddr addr, void *buf, nc::ByteSize size) const
+    {
+        if(DbgMemRead(addr, (unsigned char*)buf, size))
+            return size;
+        else
+            return 0;
+    }
+};
+
+class DbgDemangler: public nc::core::mangling::Demangler {
```
Comment (yegord) [inline src/x64dbg-module/SnowmanView.cpp]: BTW, you probably do not need this class: we have a default demangler that does something sensible for common cases (GCC, MSVC).

File: src/x64dbg-module/SnowmanView.cpp [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+    image->setDemangler(std::make_unique<DbgDemangler>());
+
+    //create sections
+    auto section = std::make_unique<nc::core::image::Section>(".text", base, size);
+    section->setReadable(true);
+    section->setWritable(true);
+    section->setExecutable(true);
+    section->setCode(true);
+    section->setData(true);
+    section->setBss(true);
```
Comment (yegord) [inline src/x64dbg-module/SnowmanView.cpp]: Should be false.
Actually, this does not influence anything currently, although it did influence before.

Comment (mrexodia) [inline src/x64dbg-module/SnowmanView.cpp]: Thanks for your comments, it is appreciated!

**Pull request closed without merging**